documentation/src/main/asciidoc/topics/proc_adding_protostream_annotations.adoc

```diff
@@ -0,0 +1,54 @@
+[id='adding-protostream-annotations_{context}']
+= Adding ProtoStream annotations to Java objects
+
+Declare Protobuf metadata by adding minimal annotations to a Java class and its fields.
+{brandname} then uses the ProtoStream processor to generate Protobuf schema and related marshallers from those annotations.
+
+.Procedure
+
+. Annotate the Java objects that you want to marshall with `@ProtoField`.
+. Use `@ProtoFactory` to annotate constructors that must include all fields for immutable objects.
++
+[NOTE]
+====
+* If no `@ProtoFactory` annotated constructor is found the class needs a default no-argument constructor, otherwise errors occur during compilation.
+
+* Field names and parameters of the `@ProtoFactory` constructor or method must match the corresponding Protobuf message. The order is not important.
+
+* Use the `@ProtoField(type)` element to change the Protobuf type, similarly to changing Java `int` to `fixed32`. Any incompatible declarations for the Java property cause compiler errors.
+====
++
+. Annotate members of any Java enum with `@ProtoEnumValue`.
+
+The following `Author.java` and `Book.java` examples show Java classes annotated with `@ProtoField` and `@ProtoFactory`:
+
+.Author.java
+[source,java,options="nowrap",subs=attributes+]
+----
+include::code_examples/Author.java[]
```

Comment:
It's no longer necessary to have the `number =` if only a single field exists on the annotation, so the example can be updated to:

```java
   @ProtoField(1)
   final String name;

   @ProtoField(2)
   final String surname;
```